neon_mq_connector/consumers/select_consumer.py

```diff
@@ -271,9 +273,7 @@ def _close_connection(self, mark_consumer_as_dead: bool = True):
 
     def reconnect(self, wait_interval: int = 5):
         self._close_connection(mark_consumer_as_dead=False)
-        # TODO: Find a better way to wait for shutdown/server restart. This will
-        #   fail to reconnect if the server isn't back up within `wait_interval`
-        time.sleep(wait_interval)
+        threading.Event().wait(wait_interval)
```

Comment:
Not sure if there is any benefit of using `Event` here since it is initialised in place and not point to any state

Reply:
`Event.wait()` is generally more efficient because it releases the GIL; `sleep` will occupy a thread for the duration of the sleep